backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/graphicpromotion/internal/BpkGraphicPromoImpl.kt

```diff
@@ -0,0 +1,287 @@
+            )
+        } else {
+            Box(
+                modifier = Modifier
+                    .matchParentSize()
+                    .indication(interactionSource, InteractiveBackgroundIndication),
+                content = image,
+            )
+            ForegroundContent(
+                headline = headline,
+                kicker = kicker,
+                subHeadline = subHeadline,
+                variant = variant,
+                verticalAlignment = verticalAlignment,
+                sponsor = sponsor,
+                sponsorLogo = sponsorLogo,
+            )
+        }
+    }
+}
+
+@Composable
+private fun SponsorOverlayView(
+    textColor: Color,
+    sponsor: BpkGraphicsPromoSponsor?,
+    modifier: Modifier = Modifier,
+    sponsorLogo: (@Composable () -> Unit)? = null,
+) {
+    if (sponsor != null) {
+        Column(
+            modifier = modifier,
+            verticalArrangement = Arrangement.spacedBy(BpkSpacing.Md, Alignment.Top),
+        ) {
+            BpkText(
+                text = sponsor.title,
+                style = BpkTheme.typography.label1,
+                color = textColor,
+            )
+
+            Box(
+                modifier = Modifier
+                    .heightIn(0.dp, SPONSOR_LOGO_HEIGHT.dp),
+                content = { sponsorLogo?.let { it() } },
+            )
+        }
+    }
+}
+
+@Composable
+private fun MessageOverlay(
+    headline: String,
+    kicker: String?,
+    subHeadline: String?,
+    textColor: Color,
+    modifier: Modifier = Modifier,
+) {
+    Column(
+        modifier = modifier,
+        verticalArrangement = Arrangement.spacedBy(BpkSpacing.Md, Alignment.Top),
+    ) {
+        if (!kicker.isNullOrBlank()) {
+            BpkText(
+                text = kicker,
+                style = BpkTheme.typography.subheading,
+                color = textColor,
+            )
+        }
+
+        BpkText(
+            text = headline,
+            style = BpkTheme.typography.heading2,
+            color = textColor,
+        )
+        if (!subHeadline.isNullOrBlank()) {
+            BpkText(
+                text = subHeadline,
+                style = BpkTheme.typography.heading5,
+                color = textColor,
+            )
+        }
+    }
+}
+
+@Composable
+private fun ForegroundContent(
+    headline: String,
+    modifier: Modifier = Modifier,
+    kicker: String? = null,
+    subHeadline: String? = null,
+    variant: BpkGraphicPromoVariant = BpkGraphicPromoVariant.OnDark,
+    verticalAlignment: BpkGraphicPromoVerticalAlignment = BpkGraphicPromoVerticalAlignment.Top,
+    sponsor: BpkGraphicsPromoSponsor? = null,
+    sponsorLogo: (@Composable () -> Unit)? = null,
+) {
+    val textColor: Color = when (variant) {
+        BpkGraphicPromoVariant.OnDark -> BpkTheme.colors.textOnDark
+        BpkGraphicPromoVariant.OnLight -> BpkTheme.colors.textOnLight
+    }
+
+    Column(modifier = Modifier.padding(BpkSpacing.Lg)) {
```

Comment:
Also, we're ignoring the modifier passed to this function – can we change this to modifier = modifier.padding() ?